### PR TITLE
Allow overriding the semver resolution engine

### LINKED ISSUE #8: Externalize SemVer parsing
Hey, very nice library. I find the concept pretty cool and I admire the simplicity behind it. I can see how it would be useful for large projects moving fast (and trying not to break too many things).

I have been looking at the code and it’s stunning as always Kaelig, so I’ll make a single comment: I think the semver parsing and version comparison should not be part of sass-deprecate. The role of this library is to emit warnings and/or to remove code based on version numbers. Not to analyse and compare versions.

I would recommend you externalize the semver parsing and version comparison into another library, kind of like what npm does with https://github.com/npm/node-semver. Then, this library would be used as a dependency and expose a set of functions to compare versions, such as `gt($version-a, $version-b)` for instance. 

There are a few benefits in doing this: 

* It would remove about 80 lines of code from sass-deprecate if I’m right, which is basically dividing the amount of code per 2.
* It would make the semver parsing and version comparison more easily testable.
* It would make the semver parsing and version comparison re-usable in other projects needing it.
* It would be easy to replace this module with an actual node-semver binding through Sassport, or anything equivalent.
* Modular code is usually better.

Tell me if you want help with this.

### PR DESCRIPTION
**Pull request by kaelig**
Closes #10
Fixes #8 

@HugoGiraudel let me know how that looks and if that solves the issue you raised.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,6 +67,26 @@ And the compiled CSS won't include `.button`:
 .button-new { background: red; border: 3px solid blue; }
 ```
 
+## Advanced Semantic Versioning Support
+
+Need to compare version numbers such as `3.2.1-beta.5` and `1.2.3-alpha.2`?
+
+By default, sass-deprecate only compares `$version` with `$app-version` in the form of `Major.Minor.Patch` (e.g. `1.2.3` with `2.0.0`).
+
+For advanced Semver support in the form of `Major.Minor.Patch-beta/alpha/rc.1`, define a `deprecate-version-greater-than($v1, $v2)` function, or rely on Hugo's [sass-semver](https://raw.githubusercontent.com/HugoGiraudel/sass-semver):
```
Comment (HugoGiraudel) [inline README.md]: SemVer*

Comment (HugoGiraudel) [inline README.md]: Hugo’s

File: README.md [hunk 2]
```diff
@@ -67,6 +67,26 @@ And the compiled CSS won't include `.button`:
...
+
+```scss
+// Override the default Semver resolution engine
```
Comment (HugoGiraudel) [inline README.md]: SemVer

File: README.md [hunk 3]
```diff
@@ -67,6 +67,26 @@ And the compiled CSS won't include `.button`:
...
+// with sass-semver: https://github.com/HugoGiraudel/sass-semver
+@import 'node_modules/sass-semver/index';
+
+@function deprecate-version-greater-than($version, $app-version) {
+  @return call(gt, $v1: $version, $v2: $app-version);
```
Comment (HugoGiraudel) [inline README.md]: Maybe use `gt(..)` rather than `call(gt, )`.

**Comment by HugoGiraudel** [comment]
Added a few comments. Looks good. :)

**Comment by kaelig** [comment]
Merci!

**Pull request merged**